cmd/containerd-shim-runhcs-v1/service_internal_podshim_test.go

```diff
@@ -723,3 +736,35 @@ func Test_PodShim_statsInternal_2ndTaskID_Success(t *testing.T) {
 		})
 	}
 }
+
+func Test_PodShim_shutdownInternal(t *testing.T) {
+	for _, now := range []bool{true, false} {
+		t.Run(fmt.Sprintf("%s_Now_%t", t.Name(), now), func(t *testing.T) {
+			s, _, _, _ := setupPodServiceWithFakes(t)
+
+			if s.IsShutdown() {
+				t.Fatal("service prematurely shutdown")
+			}
+
+			_, err := s.shutdownInternal(context.Background(), &task.ShutdownRequest{
+				ID:  s.tid,
+				Now: now,
+			})
+			if err != nil {
+				t.Fatalf("could not shut down service: %v", err)
+			}
+
+			tm := time.NewTimer(5 * time.Millisecond)
+			select {
+			case <-tm.C:
+				t.Fatalf("shutdown channel did not close")
+			case <-s.Done():
+				tm.Stop()
+			}
+
+			if !s.IsShutdown() {
+				t.Fatal("service did not shutdown")
+			}
```

Comment:
If the service isn't shutdown then we will already bail the test when timer expires. Do we still need this here?

Reply:
I didnt want to assume that `IsShutdown()` uses the same mechanism as `Done()`, as in, it could be changed to check a flag or something, so that test is to check against `Done` returning but `IsShutdown` reporting false.

Reply:
Okay, we can keep this then.